### PR TITLE
F-71A: Automation of test_40638 and test_40639

### PR DESCRIPTION
**Pull request by AshwiniS-Borse**
# Problem Statement
- Automation of test_40638 and test_40639
# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [x] Attach test execution logs
-  [x] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by AshwiniS-Borse** [comment]
@kachhwahadivya @RajeshPratapSInghChouhan Please review.

File: tests/csm/rest/test_capacity_quota.py [hunk 1]
```diff
@@ -168,14 +180,30 @@ def create_bucket_to_upload_parts(
...
+            file_size,
+            b_size):
+        """Create bucket, initiate multipart upload and upload parts."""
+        self.log.info("Initiating multipart upload")
+        if os.path.exists(self.mp_obj_path):
+            os.remove(self.mp_obj_path)
+        create_file(self.mp_obj_path, file_size, b_size=b_size)
+
+        res = self.s3_mp_test_obj.create_multipart_upload(bucket_name, object_name)
+        assert_utils.assert_true(res[0], res[1])
```
Comment (RajeshPratapSInghChouhan) [inline tests/csm/rest/test_capacity_quota.py:204]: use assert only

File: tests/csm/rest/test_capacity_quota.py [hunk 2]
```diff
@@ -23,6 +23,7 @@
 import logging
 import math
 import os
+from re import S
 import time
```
Comment (RajeshPratapSInghChouhan) [inline tests/csm/rest/test_capacity_quota.py:27]: please check

File: tests/csm/rest/test_capacity_quota.py [hunk 3]
```diff
@@ -448,70 +475,96 @@ def test_40638(self):
         """
         test_case_name = cortxlogging.get_frame()
         self.log.info("##### Test started -  %s #####", test_case_name)
-        test_cfg = self.csm_conf["test_40638"]
-        enabled = test_cfg["enabled"]
-        max_size = test_cfg["max_size"]
-        max_objects = test_cfg["max_objects"]
-        self.log.info("Step 3: Start Multipart upload S3 operations of X Mb"
-                      "X=(max_size+x)")
+        test_case_name = cortxlogging.get_frame()
```
Comment (RajeshPratapSInghChouhan) [inline tests/csm/rest/test_capacity_quota.py:478]: please check

File: tests/csm/rest/test_capacity_quota.py [hunk 4]
```diff
@@ -448,70 +475,96 @@ def test_40638(self):
...
+                "Status code check failed for get capacity"
+        
+        uid = resp.json()["capacity"]["s3"]["users"][0]["id"]  
+        t_obj = resp.json()["capacity"]["s3"]["users"][0]["objects"]
+        t_size = resp.json()["capacity"]["s3"]["users"][0]["used"]
+        m_size = resp.json()["capacity"]["s3"]["users"][0]["used_rounded"]
+        assert_utils.assert_equals(self.uid, uid, "id is not equal")
+        assert_utils.assert_equals(test_cfg["total_parts"], t_obj, 
+                                   "Number of objects not equal")
+        assert_utils.assert_equal(test_cfg["file_size"]* test_cfg["b_size"], t_size,
```
Comment (RajeshPratapSInghChouhan) [inline tests/csm/rest/test_capacity_quota.py:527]: assert_equals

**Comment by AshwiniS-Borse** [comment]
@VikasKumarC  @DhobaleVishal please review this PR.
I found one defect in multipart library.

File: libs/s3/s3_multipart_test_lib.py [hunk 5]
```diff
@@ -132,7 +132,7 @@ def upload_parts(self, mpu_id: int = None, bucket_name: str = None, object_name:
         :return: (Boolean, List of uploaded parts).
         """
         try:
-            b_size = kwargs.get("block_size", "1M")
+            b_size = int(kwargs.get("block_size", "1048576"))
```
Comment (kachhwahadivya) [inline libs/s3/s3_multipart_test_lib.py:135]: @DhobaleVishal : Can u confirm this change will not break any S3 tests?

File: tests/csm/rest/test_capacity_quota.py [hunk 6]
```diff
@@ -108,14 +110,19 @@ def setup_method(self):
...
         self.bucket = "iam-user-bucket-" + str(int(time.time_ns()))
         self.display_name = "iam-display-name-" + str(int(time.time_ns()))
         self.obj_name_prefix = "created_obj"
         self.obj_name = f'{self.obj_name_prefix}{time.perf_counter_ns()}'
         self.log.info("Verify Create bucket: %s", self.bucket)
         assert s3_misc.create_bucket(self.bucket, self.akey, self.skey), "Failed to create bucket."
         self.buckets_created.append([self.bucket, self.akey, self.skey])
+        self.s3_mp_test_obj = S3MultipartTestLib(access_key=self.akey, secret_key=self.skey,
+                                                 endpoint_url=S3_CFG["s3_url"])
+        self.s3_mp_obj = Multipart(access_key=self.akey, secret_key=self.skey,
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_capacity_quota.py:124]: U dont need this object as s3_mp_test_obj inherits Multipart.

File: tests/csm/rest/test_capacity_quota.py [hunk 7]
```diff
@@ -108,14 +110,19 @@ def setup_method(self):
         assert resp1[0], resp1[1]
         self.akey = resp.json()["keys"][0]["access_key"]
         self.skey = resp.json()["keys"][0]["secret_key"]
-        self.s3t_obj = S3TestLib(access_key=self.akey, secret_key=self.skey)
+        self.s3t_obj = S3TestLib(access_key=self.akey, secret_key=self.skey,
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_capacity_quota.py:113]: U dont need this object as s3_mp_test_obj inherits S3libs.

**Review by kachhwahadivya — APPROVED** [review]

**Comment by AshwiniS-Borse** [comment]
[TEST-40638_TestCapacityQuota_20220629050211.log](https://github.com/Seagate/cortx-test/files/9009888/TEST-40638_TestCapacityQuota_20220629050211.log)
[TEST-40639_TestCapacityQuota_20220629050308.log](https://github.com/Seagate/cortx-test/files/9009889/TEST-40639_TestCapacityQuota_20220629050308.log)
[pytest_collection_logs.txt](https://github.com/Seagate/cortx-test/files/9009890/pytest_collection_logs.txt)

File: libs/s3/s3_multipart_test_lib.py [hunk 8]
```diff
@@ -145,7 +145,7 @@ def upload_parts(self, mpu_id: int = None, bucket_name: str = None, object_name:
             with open(multipart_obj_path, "rb") as file_pointer:
                 i = 1
                 while True:
-                    data = file_pointer.read(1048576 * single_part_size)
```
Comment (DhobaleVishal) [inline libs/s3/s3_multipart_test_lib.py:148]: Reading chunk from file is not depend on b_size as this is the actual size so this change will impact all tests wherever this  method being used also ur reading more than file_size in one chunk which is not correct in case of multipart upload. I would recomend, you should revert this change.

File: libs/s3/s3_multipart_test_lib.py (hunk 5 above)
Comment (DhobaleVishal) [inline libs/s3/s3_multipart_test_lib.py:135]: This change will impact all multipart tests as this variable is introduced recently by @aditipande96 and her changes looks good to me as those are not impacting s3 tests. We can go with ur changes as well but then you need to do RVF for all(~30+) tests getting impacted because of this change.

**Review by DhobaleVishal — CHANGES_REQUESTED** [review]
@AshwiniS-Borse Added inline comments for s3 lib changes.

File: libs/s3/s3_multipart_test_lib.py (hunk 8 above)
Comment (AshwiniS-Borse) [reply libs/s3/s3_multipart_test_lib.py]: Okay. Make sense.
I will revert the changes

File: libs/s3/s3_multipart_test_lib.py (hunk 5 above)
Comment (AshwiniS-Borse) [reply libs/s3/s3_multipart_test_lib.py]: Reverted changes.

**Review by ravindrachoudhari — APPROVED** [review]
